### ISSUE TITLE
[BUG] bad argument to internal function

### ISSUE THREAD

**Issue by trouver** [dashingsoft/pyarmor#1493]
There is a `foo.py` file with the following content:

```python
def foo():
    print(1)


if __name__ == '__main__':
    foo()
```

After encrypting it with the following command, running `python foo.py` results in an error:

```sh
pyarmor gen --enable-jit --mix-str \
    --obf-code 2 --outer -O ./ \
    --platform windows.x86_64,linux.x86_64 \
    -r foo.py
```

The error message is as follows:

```
Traceback (most recent call last):
  File "./foo.py", line 2, in <module>
    from pyarmor_runtime_006032 import __pyarmor__
  File "./pyarmor_runtime_006032/__init__.py", line 49, in <module>
    __pyarmor__ = __pyarmor__().__pyarmor__
  File "./pyarmor_runtime_006032/__init__.py", line 48, in __pyarmor__
    return __import__(name, globals(), locals(), ['__pyarmor__'], level=1)
SystemError: Objects/listobject.c:240: bad argument to internal function
```
Encrypted environment
OS: CentOS 7
Python 3.10.12
pyarmor 8.3.4

Encrypted runtime environment:
OS: CentOS 7
Python 3.10.12
and
OS: Windows 8.1
Python 3.10.11

**Comment by jondy** [comment]
First try native platform obfuscation.

If it works, make clean uninstallation, and install pyarmor again.

**Comment by trouver** [comment]
After I uninstalled and reinstalled, the license became a trial version. I used the registration file to re-register, but the following error occurred:

```
INFO     Python 3.10.12
INFO     Pyarmor 8.3.8 (trial), 000000, non-profits
INFO     Platform linux.x86_64
INFO     register "~/.pyarmor/pyarmor-device-regfile-6032.4.zip"
ERROR    could not get docker host machine id
Traceback (most recent call last):
  File "./pyarmor_env/lib/python3.10/site-packages/pyarmor/cli/register.py", line 194, in _get_docker_hostname
    s.connect((host, port))
socket.gaierror: [Errno -2] Name or service not known
ERROR    could not get docker host machine id
```

**Comment by jondy** [comment]
About group license issue, please check pin issue to find solution.

You can just test it with trial version for this issue.

**Comment by jondy** [comment]
If native platform works, it means wrong cross-platform package used.

Please check installation documentation to install the right cross-platform packages.

**Comment by trouver** [comment]
I uninstalled all related packages using `pip uninstall -y pyarmor pyarmor.cli.core pyarmor.cli.core.linux pyarmor.cli.core.windows`, and reinstalled the latest version: pyarmor==8.3.8. Then I used `pyarmor gen --enable-jit --mix-str --obf-code 2 --outer -O ./ --platform windows.x86_64,linux.x86_64 -r foo.py` to re-obfuscate, but the same error still occurs.

During obfuscation, pyarmor automatically installed pyarmor.cli.core.linux==4.3.5 and pyarmor.cli.core.windows==4.3.5. If this is incorrect, could you please tell me which version of the package I should install?

**Comment by jondy** [comment]
Could you do some tests something like

1. native platform, it works or not.
2. do not use extra options, like `--mix-str`, report the minimum options which could reproduce this issue
3. just try only one platform `--platform linux.x86_64` or `--platform windows.x86_64`
4. report the full logs with debug option `pyarmor -d gen ...` if it doesn't work.

Generally I guess the problem is that using the wrong extension `pyarmor_runtime.so`

**Comment by trouver** [comment]
Hello, after my testing, I believe the problem lies with `--outer`. When I use `pyarmor gen --platform windows.x86_64,linux.x86_64 foo.py` for obfuscation, the code can run normally on Linux and Windows. However, when I use `pyarmor -d gen --outer --platform windows.x86_64,linux.x86_64 foo.py` for obfuscation, it will report an error `bad argument to internal function`. Below is the complete information when using `pyarmor -d gen --outer --platform windows.x86_64,linux.x86_64 foo.py` for obfuscation.

```
INFO     Python 3.10.12
INFO     Pyarmor 8.3.8 (group), 006xxx, xxxxx
INFO     Platform linux.x86_64
DEBUG    native platform linux.x86_64
DEBUG    home path: /home/pyenv/.pyarmor
INFO     use runtime platforms: ['windows.x86_64', 'linux.x86_64']
DEBUG    command options: {'no_runtime': False, 'platforms': ['windows.x86_64', 'linux.x86_64'], 'outer': True, 'inputs': ['foo.py']}
DEBUG    install plugin: CodesignPlugin
DEBUG    install plugin: DarwinUniversalPlugin
INFO     search inputs ...
INFO     find script foo.py
INFO     find 1 top resources
INFO     start to generate runtime files
INFO     target platforms {'linux.x86_64', 'windows.x86_64'}
INFO     write dist/pyarmor_runtime_006032/linux_x86_64/pyarmor_runtime.so
INFO     patch runtime file
DEBUG    patching runtime data at 747712
DEBUG    patch runtime file OK
DEBUG    call post runtime plugin <class 'pyarmor.cli.plugin.CodesignPlugin'>
INFO     write dist/pyarmor_runtime_006032/windows_x86_64/pyarmor_runtime.pyd
INFO     patch runtime file
DEBUG    patching runtime data at 515104
DEBUG    patch runtime file OK
DEBUG    call post runtime plugin <class 'pyarmor.cli.plugin.CodesignPlugin'>
INFO     generate runtime files OK
INFO     start to obfuscate scripts
INFO     process resource "foo"
INFO     obfuscating file foo.py
DEBUG    process inline marker
DEBUG    parse script
DEBUG    process co
DEBUG    patch co
INFO     write dist/foo.py
INFO     obfuscate scripts OK
DEBUG    call post build plugin <class 'pyarmor.cli.plugin.DarwinUniversalPlugin'>
```

At the same time, my original obfuscation command can run normally without `--outer`.

**Comment by trouver** [comment]
When I use `pyarmor gen --outer foo.py` or `pyarmor gen --platform linux.x86_64 --outer foo.py`, it can run normally.

**Comment by trouver** [comment]
Is it confirmed that this issue is a bug? How should I resolve it so that I can use `--outer` and `--platform windows.x86_64,linux.x86_64` at the same time?

**Comment by jondy** [comment]
I'll check it today

**Comment by jondy** [comment]
Reproduced, it will be fixed in next release.

**Comment by jondy** [comment]
Fixed in 8.3.10
